runtime/rastrace/method_trace.c

```diff
@@ -477,10 +477,19 @@ traceMethodArgObject(J9VMThread *thr, UDATA* arg0EA, char* cursor, UDATA length)
 		J9Class* clazz = J9OBJECT_CLAZZ(thr, object);
 		J9ROMClass * romClass = clazz->romClass;
 		J9UTF8* className = J9ROMCLASS_CLASSNAME(romClass);
+		J9JavaVM *vm = thr->javaVM;
 
-		/* TODO: handle arrays */
+		if (clazz == J9VMJAVALANGSTRING_OR_NULL(vm)) {
+			/* string arg */
+			char stringArgBuffer[J9_ARRAY_DIMENSION_LIMIT];
```

Comment:
Dont use `J9_ARRAY_DIMENSION_LIMIT`, you should define some =thing smaller like 32. Keep in mind that this can be printed out to the console.

Reply:
We may also want to add a cmdline option to vary the size of the string that is printed